### ISSUE TITLE
bloc_test whenListen should mock state

### ISSUE THREAD

**Issue by Albert221** [felangel/bloc#1535]
`whenListen` should also mock `Cubit.state`.

```dart
// selected_entry_cubit.dart

  void select(String entryId) {
    emit(_entriesCubit.state.firstWhere((entry) => entry.id == entryId));
  }
```

Tests for the method above fail because 

```
NoSuchMethodError: The method 'firstWhere' was called on null.
```

despite the `whenListen` in `blocTest`'s `build` parameter:

```dart
// selected_entry_cubit_test.dart

  build: () {
    final entriesCubit = MockEntriesCubit();
    whenListen(entriesCubit, Stream.value([entry]));
    cubit = SelectedEntryCubit(entriesCubit);

    return cubit;
  },
  act: (cubit) async => cubit.select(entry.id),
```

**Comment by felangel** [comment]
Hi @Albert221 👋 
Thanks for opening an issue!

Are you able to provide a complete sample app which reproduces the issue? Thanks so much! 🙏

**Comment by ncuillery** [comment]
I believe that's the same issue as I have. The state of the mock is null unless I listen to it.

My BlocA has a BlocB as dependency. BlocA uses the state of BlocB but doesn't listen to it (because BlocB is loaded once and for all very early in my app lifecycle).

My test of BlocA mocks BlocB, but the state of BlocB is null during the test even if I configure it with `whenListen`. 

The workaround I found: listening to BlocB for nothing:

```dart
BlocA(this.blocB) : assert(blocB != null), super(InitialBlocAState()) {
  blocB.listen((_) {});
}
```
With this ☝️ the `whenListen` instruction is effective.

**Comment by ncuillery** [comment]
Actually I think I found a solution. Instead of using `whenListen`, I use the Mockito API to mock the state directly:

```dart
when(_mockBlocB.state).thenAnswer(
  (_) => BlocBState(),
);
```

**Comment by felangel** [comment]
This is addressed as part of v8.0.0-nullsafety you can do:

```dart
whenListen(
  myBloc,
  Stream.fromIterable([...]),
  initialState: MyState.initial(),
);
```
